### PR TITLE
FEATURE: ONE-4619 Add turn off lines option

### PR DESCRIPTION
**Pull request by fiedlr**
---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/466/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2059/ : FAILURE in 9m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/469/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2062/ : FAILURE in 39m 10s

File: libs/sdk-ui-charts/prepublish.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+fswatch -r src/ | (while read;
+do
+ echo "Running prepublish.."
+ npm run prepublishOnly
+ cp -r ./dist ~/gd/gdc-analytical-designer/node_modules/\@gooddata/sdk-ui-charts
+ echo "Done!\n"
+done)
```
Comment (lupko) [inline libs/sdk-ui-charts/prepublish.sh:8]: please read docs/sdk-dev.md ; if you want to test stuff in AD/KD, you can use `yarn link`. combined with `dev` compilation mode this addresses what this script does.

note though, that when linking sdk-ui-charts, you need to also link the sdk-ui-ext (plug viz impl which also depends on the charts).

Comment (lupko) [reply libs/sdk-ui-charts/prepublish.sh:8]: additionally, I suggest to use the storybook in `sdk-ui-tests` for dev testing. that way you are confined to SDK-only for starters. when dev-testing like this within the monorepo, all the dependent packages are fully symlinked. so you all you need to do is run 'dev' compilation mode on charts

Comment (fiedlr) [reply libs/sdk-ui-charts/prepublish.sh:8]: Sorry the prepublish was committed by mistake. Thanks for the hint, though! The script pretty much works, but your solution surely seems like much better, I'll try it.

Comment (fiedlr) [reply libs/sdk-ui-charts/prepublish.sh:8]: When it comes to testing in storybooks, the feature I'm implementing is actually for AD, even though most of the code is in SDK.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/514/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2106/ : FAILURE in 8m 56s

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pushData: (data: any) => any;
+    properties: IVisualizationProperties;
+    isDisabled: boolean;
+    showDisabledMessage?: boolean;
+    defaultValue?: string | boolean;
+}
+
+class DataPointsControl extends React.Component<IDataPointsControlProps & WrappedComponentProps> {
+    public static defaultProps = {
+        defaultValue: "auto",
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:21]: in story screenshot shows default Auto, but description mentions default Show. Please unify it in story and in code

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// (C) 2019 GoodData Corporation
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:1]: 2020 :)

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public static defaultProps = {
+        defaultValue: "auto",
+        showDisabledMessage: false,
+    };
+    public render() {
+        const { pushData, properties, intl, isDisabled, showDisabledMessage, defaultValue } = this.props;
+        const dataPoints = get(properties, "controls.dataPoints.visible", defaultValue);
+
+        return (
+            <div className="s-data-labels-config">
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:29]: use unique s class. This one is already used ;)

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx [hunk 5]
```diff
@@ -38,7 +42,14 @@ export default class LineChartBasedConfigurationPanel extends BaseChartConfigura
                             pushData={pushData}
                             properties={properties}
                             isDisabled={controlsDisabled}
-                            defaultValue={false}
+                            defaultValue={"auto"}
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:45]: this change of default value is on purpose?

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx [hunk 6]
```diff
@@ -14,12 +15,15 @@ import {
 } from "../../constants/bubble";
 import BaseChartConfigurationPanel from "./BaseChartConfigurationPanel";
 
+export const CHARTS_SUPPORTING_DATAPOINT_CTRL = ["line", "area"];
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:18]: why is this enum here when it is not used in this file?

File: libs/sdk-ui-ext/src/internal/constants/supportedProperties.ts [hunk 7]
```diff
@@ -76,9 +77,13 @@ export const COLUMN_CHART_SUPPORTED_PROPERTIES = {
 };
 
 export const LINE_CHART_SUPPORTED_PROPERTIES = {
-    [AXIS.PRIMARY]: BASE_CHART_SUPPORTED_PROPERTIES,
+    [AXIS.PRIMARY]: [...BASE_CHART_SUPPORTED_PROPERTIES, "dataPoints.visible"],
     [AXIS.SECONDARY]: [...BASE_PROPERTIES, ...BASE_X_AXIS_PROPERTIES, ...BASE_SECONDARY_AXIS_PROPERTIES],
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/constants/supportedProperties.ts:81]: why is not the new control also on secondary axis?

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

File: libs/sdk-ui-charts/prepublish.sh (hunk 1 above)
Comment (lupko) [reply libs/sdk-ui-charts/prepublish.sh:8]: - you should test the configuration panel in storybook (see existing tests for inspiration).
- you should test chart configuration & updated chart behavior (react component) in storybook
- the testing infrastructure will generate insights (fka vis objects)
- stories will be automatically created to render insights using plug viz; this rendered insight will include all configuration panels with their setting + the actual render of the chart using plug viz

==> you can test probably 80+% in SDK and then do final tuning in AD.

if you have not done so, I encourage to check out sdk-ui-tests, its README and start the storybook to see what kind of stories we have.

with v8 we are aiming for greater encapsulation of SDK & its testing. ideally you should be able to verify everything in SDK and then 'just' do final verification in AD; this is not the case yet but you should be able to get quite far. besides, this particular feature probably needs 0 changes in AD codebase itself right?

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx (hunk 5 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:45]: By accident, I'll revert it.

File: libs/sdk-ui-ext/src/internal/constants/supportedProperties.ts (hunk 7 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/constants/supportedProperties.ts:81]: That's actually why I need to consult this. I was following how dataLabels were implemented and assumed primary axis is for the y axis? But I guess I assumed wrong.

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx (hunk 6 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:18]: By accident from previous implementation. Will remove it.

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx (hunk 4 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:29]: Right, I just copied this from data labels ctrl :-)

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx (hunk 2 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:21]: This is what I need to investigate, if the default behavior is right now (without the feature) really show or not.

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx (hunk 3 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationControls/DataPointsControl.tsx:1]: I guess the date updating script doesn't work on new files :-D

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-ui-charts/prepublish.sh (hunk 1 above)
Comment (fiedlr) [reply libs/sdk-ui-charts/prepublish.sh:8]: @lupko Yes, I didn't need to change anything in AD. Ok, I'll check the stories, I didn't know there were stories for configuration panels.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/614/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2204/ : FAILURE in 3m 54s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/653/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2241/ : FAILURE in 16m 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/666/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2253/ : FAILURE in 14m 17s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/672/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2258/ : FAILURE in 11m 41s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/712/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2297/ : FAILURE in 17m 31s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/715/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2300/ : FAILURE in 11m 42s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-ui-charts/src/interfaces/chartConfig.ts [hunk 8]
```diff
@@ -209,6 +215,16 @@ export type PositionType = "left" | "right" | "top" | "bottom" | "auto";
...
 
+/**
+ * Data points visibility options.
+ *
+ * - false: no points
+ * - true: datapoints shown, datapoints can overlap when rendered
+ * - "auto": datapoints shown, datapoints will not overlap when rendered
+ * @public
+ */
+export type IDataPointsVisible = boolean | string;
```
Comment (no23reason) [inline libs/sdk-ui-charts/src/interfaces/chartConfig.ts:226]: Please be more explicit:
```ts
export type IDataPointsVisible = boolean | "auto";
```

File: libs/sdk-ui-ext/src/internal/components/configurationControls/DropdownControl.tsx [hunk 9]
```diff
@@ -33,11 +34,19 @@ class DropdownControl extends React.PureComponent<IDropdownControlProps & Wrappe
...
     };
 
     constructor(props: IDropdownControlProps & WrappedComponentProps) {
         super(props);
         this.onSelect = this.onSelect.bind(this);
+        this.getSelectedItem = this.getSelectedItem.bind(this);
+
+        const { value, pushInitialValue } = props;
+
+        if (pushInitialValue) {
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/DropdownControl.tsx:47]: This should be in `componentDidMount`, side effects in constructor are not desireable.

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx [hunk 10]
```diff
@@ -1,25 +1,32 @@
 // (C) 2019 GoodData Corporation
 import React from "react";
+import has from "lodash/get";
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:3]: Is this renaming intentional? Importig `has` from `get`?

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/768/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2354/ : FAILURE in 2m 47s

**Pull request closed without merging**